test/semantics/if_arith02.f90

```diff
@@ -0,0 +1,50 @@
+! Copyright (c) 2019, NVIDIA CORPORATION.  All rights reserved.
+!
+! Licensed under the Apache License, Version 2.0 (the "License");
+! you may not use this file except in compliance with the License.
+! You may obtain a copy of the License at
+!
+!     http://www.apache.org/licenses/LICENSE-2.0
+!
+! Unless required by applicable law or agreed to in writing, software
+! distributed under the License is distributed on an "AS IS" BASIS,
+! WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+! See the License for the specific language governing permissions and
+! limitations under the License.
+
+! Check that only labels are allowed in arithmetic if statements.
+! TODO: Revisit error message "expected 'ASSIGN'" etc.
+! TODO: Revisit error message "expected one of '0123456789'"
+
+! TODO: BUG: Note that labels 500 and 600 do not exist and
+! ought to be flagged as errors. This oversight may be the
+! result of disabling semantic checking after syntax errors.
+
+if ( A ) 500, 600, 600
+100 CONTINUE
+200 CONTINUE
+300 CONTINUE
+
+!ERROR: expected 'ASSIGN'
+!ERROR: expected 'ALLOCATE ('
+!ERROR: expected '=>'
+!ERROR: expected '('
+!ERROR: expected '='
```

Comment:
I'm not sure we should be testing the parse in test/semantics.

Reply:
I didn't expect that the parser would fail.